autoscale/src/main/java/com/sequenceiq/periscope/service/ha/LeaderElectionService.java

```diff
@@ -112,8 +112,8 @@ public void run() {
                         try {
                             long limit = clock.getCurrentTimeMillis() - heartbeatThresholdRate;
                             List<PeriscopeNode> activeNodes = periscopeNodeRepository.findAllByLastUpdatedIsGreaterThan(limit);
-                            reallocateOrphanClusters(activeNodes);
                             if (!activeNodes.isEmpty()) {
+                                reallocateOrphanClusters(activeNodes);
```

Comment:
what will happen if the active nodes are empty?

Reply:
When active nodes are empty, which is possible when there is a delayed heart beat, there is no action to be performed. The next timer cycle would continue processing the allocation when nodes update their heartbeat but the timer itself is not stopped which happens now.